department-of-veterans-affairs/abd-vro
# Adds Logging for Expanded Classification Endpoint

## What was the problem?
This PR adds logging to track Endpoint in datadog logs.  The logging structure is the same as `/vagov-claim-classifier` endpoint with the exception of adding the endpoint name to differentiate between the endpoints in datadog.  The logs will look:
- Contention Log:
```
{
    "vagov_claim_id": 100,
    "claim_type": "new",
    "classification_code": 249482,
    "classification_name": "Musculoskeletal - Bones",
    "contention_text": "unmapped contention text",
    "diagnostic_code": "None",
    "is_in_dropdown": false,
    "is_lookup_table_match": true,
    "is_multi_contention": true,
    "endpoint": "/expanded-contention-classification",
    "date": "2024-11-04 13:40:25",
    "level": "info"
}
```
- Claim Log:
```
{
    "claim_id": 100,
    "form526_submission_id": 500,
    "is_fully_classified": false,
    "percent_clasified": 81.25,
    "num_processed_contentions": 16,
    "num_classified_contentions": 13,
    "endpoint": "/expanded-contention-classification",
    "date": "2024-11-04 13:40:25",
    "level": "info"
}
```

Associated tickets or Slack threads:
- [#626](https://app.zenhub.com/workspaces/contention-classification-640a24db5dcd08457cebbc79/issues/gh/department-of-veterans-affairs/vagov-claim-classification/626)

## How does this fix it?[^1]
This adds logging to the endpoint calls for both contentions and claim level logging. 

## How to test this PR
- Can test manually in Postman or other API testing tool and send test request and see the logs in the terminal. No unit tests written specifically for logging.
- Step 2[^secrel]


[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).
